### PR TITLE
srgn: init at 0.10.2

### LINKED ISSUE #293066: Package request: srgn
**Project description**

# srgn - a code surgeon

A code **s**u**rg**eo**n** for precise text and code transplantation.

Born a Unicode-capable [descendant of `tr`](https://github.com/alexpovel/srgn?tab=readme-ov-file#comparison-with-tr), `srgn` adds useful [*actions*](https://github.com/alexpovel/srgn?tab=readme-ov-file#actions), acting within precise, optionally language grammar-aware [*scopes*](https://github.com/alexpovel/srgn?tab=readme-ov-file#scopes). It suits use cases where...

- regex [doesn't cut it](https://en.wikipedia.org/wiki/Pumping_lemma_for_regular_languages) anymore,
- editor tools such as *Rename all* are too specific, and not automatable,
- precise manipulation, not just matching, is required, and lastly and optionally,
- Unicode-specific trickery is desired.

**Metadata**

* homepage URL: https://github.com/alexpovel/srgn
* source URL: https://github.com/alexpovel/srgn
* license: [MIT](https://github.com/alexpovel/srgn/blob/main/LICENSE)
* platforms: linux, darwin

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by lolbinarycat**
## Description of changes

Closes #293066

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/sr/srgn/package.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    repo = "srgn";
+    rev = "9df9fe772df140c2cd0f3c8311c1e12c210838de";
+    hash = "sha256-qUkuE0Tws7pl0Hd9Dsi6+V74ErCv+BwdvTdCbvAjgIc=";
+  };
+
+  cargoHash = "sha256-B0epxhyPoImoclQRQHVgKPxnEdb8ER2mjJgse7FtgqU=";
+
+  meta = with lib; {
+    description = "A code surgeon for precise text and code transplantation.";
+    homepage = "https://github.com/NixOS/nixpkgs/issues/293066";
```
Comment (NyCodeGHG) [inline pkgs/by-name/sr/srgn/package.nix:21]: i'm not sure this is the right website 🤔

Comment (lolbinarycat) [reply pkgs/by-name/sr/srgn/package.nix:21]: whoops, you are correct.  should be fixed now.

File: pkgs/by-name/sr/srgn/package.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    owner = "alexpovel";
+    repo = "srgn";
+    rev = "9df9fe772df140c2cd0f3c8311c1e12c210838de";
+    hash = "sha256-qUkuE0Tws7pl0Hd9Dsi6+V74ErCv+BwdvTdCbvAjgIc=";
+  };
+
+  cargoHash = "sha256-B0epxhyPoImoclQRQHVgKPxnEdb8ER2mjJgse7FtgqU=";
+
+  meta = with lib; {
+    description = "A code surgeon for precise text and code transplantation.";
```
Comment (pbsds) [inline pkgs/by-name/sr/srgn/package.nix:20]:
```suggestion
    description = "A code surgeon for precise text and code transplantation";
```

**Review by pbsds — COMMENTED** [review]
Result of `nixpkgs-review pr 293076` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>srgn</li>
  </ul>
</details>



- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package builds on x86_64-linux
- [x] executables tested on x86_64-linux
- [ ] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [x] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden
- [x] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
- [x] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [x] patches that are remotely available are fetched rather than vendored

**Comment by marcusramberg** [comment]
I think the commits should be squashed, with the init message updated to the actual version.

File: pkgs/by-name/sr/srgn/package.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "srgn";
+  # the latest versioned release has broken unit tests, so we use the latest commit instead
```
Comment (alexpovel) [inline pkgs/by-name/sr/srgn/package.nix:7]: We fixed this is https://github.com/alexpovel/srgn/issues/66 , so this wouldn't be necessary anymore.

**Comment by computerscience-person** [comment]
@thecaralice has a pull request of an updated version of srgn. Maybe theirs should be merged into main instead?

**Comment by pbsds** [comment]
done in https://github.com/NixOS/nixpkgs/pull/308346

**Pull request closed without merging**